src/services/cosmos.ts

```diff
@@ -650,7 +666,9 @@ export class Cosmos {
       params,
     );
 
-    return await this.sendMsg(transport, hdPath, sender, msg);
+    const {address: from} = await transport.getAccountInfo(hdPath);
```

Comment:
I think you need same `showModal('ledgerLocked');` logic here too, because it's the same `getAccountInfo` call